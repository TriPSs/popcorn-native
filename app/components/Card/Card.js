import React from 'react'
import PropTypes from 'prop-types'
import { StyleSheet, View } from 'react-native'

import dimensions from 'modules/dimensions'

import BaseButton from '../BaseButton'
import Overlay from '../Overlay'
import Image from '../Image'
import Container from '../Container'

export const styles = StyleSheet.create({

  root: {
    borderRadius: dimensions.BORDER_RADIUS,
    overflow: 'hidden',
    margin: dimensions.UNIT,
    height: dimensions.CARD_HEIGHT,
    width: dimensions.CARD_WIDTH,
  },

  small: {
    height: dimensions.CARD_HEIGHT_SMALL,
    width: dimensions.CARD_WIDTH_SMALL,
    margin: dimensions.UNIT / 2,
  },

})

export const Card = ({ item, variant, elevation, empty, style, ...rest }) => (
  <Container
    elevation={elevation}
    style={[
      styles.root,
      variant === 'small' && styles.small,
      style,
    ]}>
    <BaseButton
      // onLongPress={() => console.warn(item.title)}
      // onPress={() => console.log('press')}
      {...rest}>
      <View>
        <Image
          type={'poster'}
          size={'thumb'}
          images={
            empty
              ? {}
              : item.images
          } />

        {item && item.watched && item.watched.complete && (
          <Overlay variant={'dark'} />
        )}
      </View>
    </BaseButton>
  </Container>
)

Card.propTypes = {
  item: PropTypes.object,
  empty: PropTypes.bool,
  style: PropTypes.object,
  elevation: PropTypes.number,
  variant: PropTypes.oneOf(['default', 'small']),
}

Card.defaultProps = {
  item: null,
  empty: false,
  style: null,
  component: BaseButton,
  elevation: 1,
  variant: 'default',
}

export default Card
