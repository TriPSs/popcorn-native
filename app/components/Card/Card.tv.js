import React from 'react'
import PropTypes from 'prop-types'
import { StyleSheet, View, Platform } from 'react-native'

import dimensions from 'modules/dimensions'
import colors from 'modules/colors'

import BaseButton from '../BaseButton'
import Overlay from '../Overlay'
import Image from '../Image'
import Container from '../Container'

import BaseCard, { styles as baseStyles } from './Card'

export const styles = StyleSheet.create({

  root: {
    ...baseStyles.root,

    borderWidth: 2,
    borderColor: 'transparent',
    borderRadius: dimensions.BORDER_RADIUS,
  },

  card: {
    margin: 0,
    overflow: 'visible',
  },

  active: {
    borderColor: colors.PRIMARY_COLOR_200,
  },
})

export const Card = React.memo(({
  style,
  isTvSelected,
  variant,
  ...rest
}) => (
  <View style={[
    styles.root,
    baseStyles[variant],
    isTvSelected
      ? styles.active
      : {},
  ]}>
    <BaseCard
      {...rest}
      variant={variant}
      style={{
        ...baseStyles[variant],
        ...styles.card,
        ...style,
      }}
    />
  </View>
))

Card.propTypes = {
  ...BaseCard.propTypes,
}

Card.defaultProps = {
  ...BaseCard.defaultProps,
}

export default Card
