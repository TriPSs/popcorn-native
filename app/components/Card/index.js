import useCorrect from 'modules/useCorrect'

import Card from './Card'
import CardTV from './Card.tv'

export default useCorrect(Card, null, CardTV)
