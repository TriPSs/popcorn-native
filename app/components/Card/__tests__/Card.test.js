import React from 'react'
import { shallow } from 'enzyme'

import Card from '../Card'

describe('(Component) Card', () => {

  const item = {
    images: {
      poster: {
        thumb: 'test-image',
      },
    },
  }

  it('should render with defaults', () => {
    const result = shallow(
      <Card empty />,
    )

    expect(result).toMatchSnapshot()
  })

  it('should render variant small with defaults', () => {
    const result = shallow(
      <Card
        variant={'small'}
        empty />,
    )

    expect(result).toMatchSnapshot()
  })

  it('should render with images from item', () => {
    const result = shallow(
      <Card item={item} />,
    )

    expect(result).toMatchSnapshot()
  })

  it('should render a overlay if the item is watched', () => {
    const result = shallow(
      <Card
        item={{
          ...item,
          watched: {
            complete: true,
          },
        }}
      />,
    )

    expect(result).toMatchSnapshot()
  })

  it('should render a overlay if it\'s forced', () => {
    const result = shallow(
      <Card
        forceOverlay
        item={{
          ...item,
          watched: {
            complete: false,
          },
        }}
      />,
    )

    expect(result).toMatchSnapshot()
  })

  it('should not render a overlay if it\'s forced but overlayAllowed is false', () => {
    const result = shallow(
      <Card
        overlayAllowed={false}
        forceOverlay
        item={{
          ...item,
          watched: {
            complete: false,
          },
        }}
      />,
    )

    expect(result).toMatchSnapshot()
  })


})
