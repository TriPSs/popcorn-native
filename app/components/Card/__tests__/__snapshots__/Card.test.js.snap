// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Component) Card should not render a overlay if it's forced but overlayAllowed is false 1`] = `
<Container
  elevation={1}
  style={
    Array [
      Object {
        "borderRadius": 5,
        "height": 135.41666666666669,
        "margin": 8,
        "overflow": "hidden",
        "width": 93.75,
      },
      false,
      null,
    ]
  }
>
  <BaseButton
    component={[Function]}
    hide={false}
    innerRef={null}
    rippleBorderless={true}
    rippleColor="rgba(0, 0, 0, .3)"
  >
    <View>
      <Image
        empty={false}
        images={
          Object {
            "poster": Object {
              "thumb": "test-image",
            },
          }
        }
        resizeMode="stretch"
        size="thumb"
        style={null}
        type="poster"
        withFallback={true}
      />
    </View>
  </BaseButton>
</Container>
`;

exports[`(Component) Card should render a overlay if it's forced 1`] = `
<Container
  elevation={1}
  style={
    Array [
      Object {
        "borderRadius": 5,
        "height": 135.41666666666669,
        "margin": 8,
        "overflow": "hidden",
        "width": 93.75,
      },
      false,
      null,
    ]
  }
>
  <BaseButton
    component={[Function]}
    hide={false}
    innerRef={null}
    rippleBorderless={true}
    rippleColor="rgba(0, 0, 0, .3)"
  >
    <View>
      <Image
        empty={false}
        images={
          Object {
            "poster": Object {
              "thumb": "test-image",
            },
          }
        }
        resizeMode="stretch"
        size="thumb"
        style={null}
        type="poster"
        withFallback={true}
      />
      <Overlay
        style={Object {}}
        variant="dark"
        withAnimation={false}
      />
    </View>
  </BaseButton>
</Container>
`;

exports[`(Component) Card should render a overlay if the item is watched 1`] = `
<Container
  elevation={1}
  style={
    Array [
      Object {
        "borderRadius": 5,
        "height": 135.41666666666669,
        "margin": 8,
        "overflow": "hidden",
        "width": 93.75,
      },
      false,
      null,
    ]
  }
>
  <BaseButton
    component={[Function]}
    hide={false}
    innerRef={null}
    rippleBorderless={true}
    rippleColor="rgba(0, 0, 0, .3)"
  >
    <View>
      <Image
        empty={false}
        images={
          Object {
            "poster": Object {
              "thumb": "test-image",
            },
          }
        }
        resizeMode="stretch"
        size="thumb"
        style={null}
        type="poster"
        withFallback={true}
      />
      <Overlay
        style={Object {}}
        variant="dark"
        withAnimation={false}
      />
    </View>
  </BaseButton>
</Container>
`;

exports[`(Component) Card should render variant small with defaults 1`] = `
<Container
  elevation={1}
  style={
    Array [
      Object {
        "borderRadius": 5,
        "height": 135.41666666666669,
        "margin": 8,
        "overflow": "hidden",
        "width": 93.75,
      },
      Object {
        "height": 120.83333333333334,
        "margin": 4,
        "width": 84.375,
      },
      null,
    ]
  }
>
  <BaseButton
    component={[Function]}
    hide={false}
    innerRef={null}
    rippleBorderless={true}
    rippleColor="rgba(0, 0, 0, .3)"
  >
    <View>
      <Image
        empty={false}
        images={Object {}}
        resizeMode="stretch"
        size="thumb"
        style={null}
        type="poster"
        withFallback={true}
      />
    </View>
  </BaseButton>
</Container>
`;

exports[`(Component) Card should render with defaults 1`] = `
<Container
  elevation={1}
  style={
    Array [
      Object {
        "borderRadius": 5,
        "height": 135.41666666666669,
        "margin": 8,
        "overflow": "hidden",
        "width": 93.75,
      },
      false,
      null,
    ]
  }
>
  <BaseButton
    component={[Function]}
    hide={false}
    innerRef={null}
    rippleBorderless={true}
    rippleColor="rgba(0, 0, 0, .3)"
  >
    <View>
      <Image
        empty={false}
        images={Object {}}
        resizeMode="stretch"
        size="thumb"
        style={null}
        type="poster"
        withFallback={true}
      />
    </View>
  </BaseButton>
</Container>
`;

exports[`(Component) Card should render with images from item 1`] = `
<Container
  elevation={1}
  style={
    Array [
      Object {
        "borderRadius": 5,
        "height": 135.41666666666669,
        "margin": 8,
        "overflow": "hidden",
        "width": 93.75,
      },
      false,
      null,
    ]
  }
>
  <BaseButton
    component={[Function]}
    hide={false}
    innerRef={null}
    rippleBorderless={true}
    rippleColor="rgba(0, 0, 0, .3)"
  >
    <View>
      <Image
        empty={false}
        images={
          Object {
            "poster": Object {
              "thumb": "test-image",
            },
          }
        }
        resizeMode="stretch"
        size="thumb"
        style={null}
        type="poster"
        withFallback={true}
      />
    </View>
  </BaseButton>
</Container>
`;
